Annotato/annotato
# Add Fluent ORM

Changes
- Add [Fluent](https://docs.vapor.codes/4.0/fluent/overview/) ORM
- Create the `Document` model and migration that serve as a sanity check that the database is connected

Testing the database migration
1. Run psql and create a database named `annotato`
2. `cd AnnotatoBackend; cp .env.template .env.development`
3. Fill in `env.development` with your own `DB_USERNAME` and `DB_PASSWORD` (this may not be necessary)
4. In Xcode, edit the `Run` scheme for `AnnotatoBackend`
    - add an argument `migrate`. 
    - set custom directory to be the root directory of `AnnotatoBackend` (where the `Package.swift` is under) - see [here](https://docs.vapor.codes/4.0/xcode/) for more info
5. Run. This runs any migrations in `configure.swift` on the database. You will be prompted to confirm the migration by vapor.
6. Check that your database is migrated

Note: Having the `migrate` argument means that you're only migrating the database and will not be starting the server. Alternatively you can run `vapor run migrate` from terminal in the root backend dir.
